packages/expect/src/jest-extend.ts

```diff
@@ -121,6 +121,15 @@ function JestExtendPlugin(expect: ExpectStatic, matchers: MatchersObject): ChaiP
         value: (...sample: [unknown, ...unknown[]]) => new CustomMatcher(true, ...sample),
         writable: true,
       })
+
+      // keep track of asymmetric matchers on global so that it can be copied over to local context's `expect`.
+      // note that the negated variant is automatically shared since it's assigned on the single `expect.not` object.
+      Object.defineProperty(((globalThis as any)[ASYMMETRIC_MATCHERS_OBJECT]), expectAssertionName, {
+        configurable: true,
+        enumerable: true,
+        value: (...sample: [unknown, ...unknown[]]) => new CustomMatcher(false, ...sample),
```

Comment:
1. does it work with expect.not?
2. I think they should reference the same function, can we reuse it?

Reply:
> 1. does it work with expect.not?

It's very primitive, but I included a test to check `expect.not.toEqual_testCustom` is functional in `test/core/test/local-context.test.ts`, namely this part. Do you think this is sufficient?

```tsx
localExpect(0).toEqual((localExpect as any).not.toEqual_testCustom(1))
```

> 2. I think they should reference the same function, can we reuse it?

Indeed, that might be better, but that might not significantly simplify things.
I extracted the matcher function in https://github.com/vitest-dev/vitest/pull/4405/commits/093114aaf4a6e21283d1c6bd090070c3b2c3fdca Is this what you intended?

Reply:
> I extracted the matcher function in [093114a](https://github.com/vitest-dev/vitest/commit/093114aaf4a6e21283d1c6bd090070c3b2c3fdca) Is this what you intended?

Yes, this is what I meant